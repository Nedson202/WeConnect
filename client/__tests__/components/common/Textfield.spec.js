import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Textfield from '../../../components/common/TextField';

configure({ adapter: new Adapter() });

let props;

const setup = () => {
  props = {
    field: 'username',
    value: 'username',
    label: 'Username',
    placeholder: 'Do this',
    type: 'text',
    onChange: jest.fn(),
    id: 'username'
  }

  return shallow(<Textfield {...props} />);
};

describe('Component: Textfield', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(2);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('label').length).toBe(1);
    expect(wrapper.find('input').length).toBe(1);
    expect(wrapper.length).toBe(1);
  }) 
});