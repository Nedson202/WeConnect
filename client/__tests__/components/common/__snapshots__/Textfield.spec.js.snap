// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Textfield should render 1`] = `
<div>
  <div
    className="form-group col-md-6 offset-md-3 col-lg-8 offset-lg-2"
  >
    <label
      className="control-label"
      id="control-label"
    >
      Username
    </label>
    <input
      className="form-control"
      name="username"
      onChange={[MockFunction]}
      placeholder="Do this"
      type="text"
      value="username"
    />
  </div>
</div>
`;
